permissions/middleware.go

```diff
@@ -64,13 +70,57 @@ func AuthenticationMiddleware() gin.HandlerFunc {
 				}
 
 				c.Set("ID_TOKEN_CLAIMS", claims)
+
+				mdb := c.MustGet("MDB").(*sql.DB)
+				user, err := getOrCreateUser(mdb, &claims)
+				if err != nil {
+					c.AbortWithError(http.StatusBadRequest, err).SetType(gin.ErrorTypePublic)
```

Comment:
this is not a 400 bad request error.
it is a 500 internal server error. type private